### PR TITLE
check for 404 errors. Fix #69 SIGSEGV fault

### LINKED ISSUE #69: Introduce health check
Before the download process start, we should ensure that the source is reachable first. see #68

### PR DESCRIPTION
**Pull request by hakuamesan**
Fixes #69 

Just a quick check to see if it returns 404 error. At least it avoids the SIGSEGV error.

### CODE REVIEW

File: cmd/app/downloader.go [hunk 1]
```diff
@@ -85,6 +86,17 @@ func download(options *config.Options, bindLogsToChannel bool) {
 				continue
 			}
 
+			res, err := http.Get(u)
+			if err != nil {
+				log.Error("Invalid url")
```
Comment (Girbons) [inline cmd/app/downloader.go:91]:
```go
msg := "Invalid URL"
log.Error(msg)
sendToChannel(bindLogsToChannel, msg)
```

File: cmd/app/downloader.go [hunk 2]
```diff
@@ -85,6 +86,17 @@ func download(options *config.Options, bindLogsToChannel bool) {
...
+				continue
+			}
+
+			if res.StatusCode == 404 {
+				log.Error("404: URL not found")
```
Comment (Girbons) [inline cmd/app/downloader.go:96]:
```go
msg := "404: URL not found"
log.Error(msg)
sendToChannel(bindLogsToChannel, msg)
```

**Review by Girbons — COMMENTED** [review]
@hakuamesan thank you for submitting this PR! 

It looks good! What is missing is to send the logs even to the GUI app

**Comment by hakuamesan** [comment]
ah.. sorry, i have updated the code.  I was a bit careless before!

**Comment by Girbons** [comment]
@hakuamesan don't worry 😄

**Pull request merged**